merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/MissionModelProcessor.java

```diff
@@ -80,22 +82,48 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
     for (final var element : roundEnv.getElementsAnnotatedWith(MissionModel.class)) {
       final var packageElement = (PackageElement) element;
       try {
-        final var missionModelRecord = missionModelParser.parseMissionModel(packageElement);
+        final var missionModelRecord$ = missionModelParser.parseMissionModel(packageElement);
 
         final var generatedFiles = new ArrayList<>(List.of(
-            missionModelGen.generateMerlinPlugin(missionModelRecord),
-            missionModelGen.generateSchedulerPlugin(missionModelRecord)));
+            missionModelGen.generateMerlinPlugin(missionModelRecord$),
+            missionModelGen.generateSchedulerPlugin(missionModelRecord$)));
 
-        missionModelRecord.modelConfigurationType
-            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord, configType))
+        missionModelRecord$.modelConfigurationType
+            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord$, configType))
             .ifPresent(generatedFiles::add);
 
         generatedFiles.addAll(List.of(
-            missionModelGen.generateMissionModelFactory(missionModelRecord),
-            missionModelGen.generateSchedulerModel(missionModelRecord),
-            missionModelGen.generateActivityActions(missionModelRecord),
-            missionModelGen.generateActivityTypes(missionModelRecord)
+            missionModelGen.generateMissionModelFactory(missionModelRecord$),
+            missionModelGen.generateSchedulerModel(missionModelRecord$),
+            missionModelGen.generateActivityActions(missionModelRecord$),
+            missionModelGen.generateActivityTypes(missionModelRecord$)
         ));
+
+        final var autoValueMapperRequests = roundEnv.getElementsAnnotatedWith(AutoValueMapper.class).stream().map(Element::asType).toList();
+        {
+          final var recordType = elementUtils.getTypeElement("java.lang.Record").asType();
+          for (final var request : autoValueMapperRequests) {
+            if (!typeUtils.isSubtype(request, recordType)) {
+              messager.printError("AutoValueMapper is only allowed on record types. %s is not a record".formatted(request.toString()));
```

Comment:
Whenever possible, try to use the overloads of `printError` that take an `Element` as an extra argument. This associates the error message to the location of that element, making it easier for a user to jump to and fix the issue. (There are also even more precise overloads of `printMessage` that focus on an annotation, or even an attribute of an annotation.)

(As with a sibling comment, this change requires deferring the use of `Element::asType` to later, so that `request` has the right type.)
```suggestion
              messager.printError("AutoValueMapper is only allowed on record types. %s is not a record".formatted(request.toString()), request);
```

Reply:
Done. I do really like how the warning prints out with the exact code location ❤️